rllib/models/modelv2.py

```diff
@@ -238,16 +238,20 @@ def from_batch(self, train_batch: SampleBatch,
         right input dict, state, and seq len arguments.
         """
 
-        train_batch["is_training"] = is_training
+        input_dict = train_batch.copy()
+        input_dict["is_training"] = is_training
         states = []
         i = 0
-        while "state_in_{}".format(i) in train_batch:
-            states.append(train_batch["state_in_{}".format(i)])
+        while "state_in_{}".format(i) in input_dict:
+            states.append(input_dict["state_in_{}".format(i)])
             i += 1
-        ret = self.__call__(train_batch, states, train_batch.get("seq_lens"))
-        del train_batch["is_training"]
+        ret = self.__call__(input_dict, states, input_dict.get("seq_lens"))
         return ret
 
+    # TODO: (sven) Experimental method.
+    def preprocess_train_batch(self, train_batch):
```

Comment:
Can you comment on the necessity of this API? It would be preferable to not need to add new public API surface

Reply:
Yeah, thinking about this further.
- I introduced this b/c I thought that -padding RNN inputs vs attention inputs was very different (it's actually not that much different).
- More importantly: The logic should stay in the view requirements, so maybe we can leave the padding function call outside the model, but allow it to peek at the view requirements to know, what to do. 

I'll change this.